### DISCUSSION TITLE
How to make `:w !<some_command>` open a terminal buffer or other permanent window rather than a pop-up?

### DISCUSSION THREAD

**Discussion by gwenwindflower** [LazyVim/LazyVim#2779] [Q&A] [answered]
When I run a `:w bang command` the ui seems to be configured to open a pop-up, if I'm trying to read long output this isn't ideal for my specific use-cases, I'd like to configure it to open it in a more permanent window or terminal buffer. Any tips on achieving this? I've looked around a fair amount, but am still not sure where to dig in. Thanks for any help!

**Comment by dpetka2001** [comment]
Maybe `<leader>sna` or `<leader>snl` could be of help here. Otherwise just open a terminal (even the integrated `lazyterm` if you'd like, but whatever is your preference really) and run the command?

**Reply by gwenwindflower** [reply, marked as answer]
Thanks @dpetka2001 -- I should've explained my use case more -- I'm using [`mods`](https://github.com/charmbracelet/mods?tab=readme-ov-file) to get LLM output for visual mode selection input across various commands i've created and keymapped, so having it run in-context like this is kind of the point.

I figured out what I needed though just now -- it was really the length of the output that was making the pop-up un-ideal. Needed to set `long_message_to_split=true` in my noice settings.

```lua plugins.lua
{
    "folke/noice.nvim",
    opts = {
      presets = {
        long_message_to_split = true,
      },
    },
  },
```
